crates/settlement-clients/starknet/src/lib.rs

```diff
@@ -128,20 +128,26 @@ impl SettlementClient for StarknetSettlementClient {
 
     /// Should verify the inclusion of a tx in the settlement layer
     async fn verify_tx_inclusion(&self, tx_hash: &str) -> Result<SettlementVerificationStatus> {
-        let tx_hash = FieldElement::from_hex_be(tx_hash)?;
+        let tx_hash = Felt::from_hex(tx_hash)?;
         let tx_receipt = self.account.provider().get_transaction_receipt(tx_hash).await?;
-        match tx_receipt {
-            MaybePendingTransactionReceipt::Receipt(tx) => match tx.execution_result() {
-                ExecutionResult::Succeeded => Ok(SettlementVerificationStatus::Verified),
-                ExecutionResult::Reverted { reason } => {
-                    Ok(SettlementVerificationStatus::Rejected(format!("Tx {} has been reverted: {}", tx_hash, reason)))
-                }
-            },
-            MaybePendingTransactionReceipt::PendingReceipt(tx) => match tx.execution_result() {
-                ExecutionResult::Succeeded => Ok(SettlementVerificationStatus::Pending),
-                ExecutionResult::Reverted { reason } => Ok(SettlementVerificationStatus::Rejected(format!(
+        let execution_result = tx_receipt.receipt.execution_result();
+        let status = execution_result.status();
+
+        match tx_receipt.block.is_pending() {
```

Comment:
nitpick: if it's just true/false we can do do if/else instead of a match

Reply:
Yes, should be easier to read. Updated